In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv("/content/drive/MyDrive/Colab Notebooks/Titanic-Dataset.csv")

In [5]:
PassengerId = df["PassengerId"].to_numpy()
print(PassengerId)

[  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18
  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36
  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54
  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72
  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90
  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108
 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126
 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144
 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162
 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180
 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198
 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216
 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234
 235 236 237 238 239 240 241 242 243 244 245 246 24

In [18]:
survived_series = df["Survived"]
print("Survived Series (first 5 rows):\n", survived_series.head())

Survived Series (first 5 rows):
 0    0
1    1
2    1
3    1
4    0
Name: Survived, dtype: int64


In [7]:
Pclass = df["Pclass"].to_numpy()
print(Pclass)

[3 1 3 1 3 3 1 3 3 2 3 1 3 3 3 2 3 2 3 3 2 2 3 1 3 3 3 1 3 3 1 1 3 2 1 1 3
 3 3 3 3 2 3 2 3 3 3 3 3 3 3 3 1 2 1 1 2 3 2 3 3 1 1 3 1 3 2 3 3 3 2 3 2 3
 3 3 3 3 2 3 3 3 3 1 2 3 3 3 1 3 3 3 1 3 3 3 1 1 2 2 3 3 1 3 3 3 3 3 3 3 1
 3 3 3 3 3 3 2 1 3 2 3 2 2 1 3 3 3 3 3 3 3 3 2 2 2 1 1 3 1 3 3 3 3 2 2 3 3
 2 2 2 1 3 3 3 1 3 3 3 3 3 2 3 3 3 3 1 3 1 3 1 3 3 3 1 3 3 1 2 3 3 2 3 2 3
 1 3 1 3 3 2 2 3 2 1 1 3 3 3 2 3 3 3 3 3 3 3 3 3 1 3 2 3 2 3 1 3 2 1 2 3 2
 3 3 1 3 2 3 2 3 1 3 2 3 2 3 2 2 2 2 3 3 2 3 3 1 3 2 1 2 3 3 1 3 3 3 1 1 1
 2 3 3 1 1 3 2 3 3 1 1 1 3 2 1 3 1 3 2 3 3 3 3 3 3 1 3 3 3 2 3 1 1 2 3 3 1
 3 1 1 1 3 3 3 2 3 1 1 1 2 1 1 1 2 3 2 3 2 2 1 1 3 3 2 2 3 1 3 2 3 1 3 1 1
 3 1 3 1 1 3 1 2 1 2 2 2 2 2 3 3 3 3 1 3 3 3 3 1 2 3 3 3 2 3 3 3 3 1 3 3 1
 1 3 3 1 3 1 3 1 3 3 1 3 3 1 3 2 3 2 3 2 1 3 3 1 3 3 3 2 2 2 3 3 3 3 3 2 3
 2 3 3 3 3 1 2 3 3 2 2 2 3 3 3 3 3 3 3 2 2 3 3 1 3 2 3 1 1 3 2 1 2 2 3 3 2
 3 1 2 1 3 1 2 3 1 1 3 3 1 1 2 3 1 3 1 2 3 3 2 1 3 3 3 3 2 2 3 1 2 3 3 3 3
 2 3 3 1 3 1 1 3 3 3 3 1 

In [8]:
name = df["Name"].to_numpy()
print(name)

['Braund, Mr. Owen Harris'
 'Cumings, Mrs. John Bradley (Florence Briggs Thayer)'
 'Heikkinen, Miss. Laina' 'Futrelle, Mrs. Jacques Heath (Lily May Peel)'
 'Allen, Mr. William Henry' 'Moran, Mr. James' 'McCarthy, Mr. Timothy J'
 'Palsson, Master. Gosta Leonard'
 'Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)'
 'Nasser, Mrs. Nicholas (Adele Achem)' 'Sandstrom, Miss. Marguerite Rut'
 'Bonnell, Miss. Elizabeth' 'Saundercock, Mr. William Henry'
 'Andersson, Mr. Anders Johan' 'Vestrom, Miss. Hulda Amanda Adolfina'
 'Hewlett, Mrs. (Mary D Kingcome) ' 'Rice, Master. Eugene'
 'Williams, Mr. Charles Eugene'
 'Vander Planke, Mrs. Julius (Emelia Maria Vandemoortele)'
 'Masselmani, Mrs. Fatima' 'Fynney, Mr. Joseph J' 'Beesley, Mr. Lawrence'
 'McGowan, Miss. Anna "Annie"' 'Sloper, Mr. William Thompson'
 'Palsson, Miss. Torborg Danira'
 'Asplund, Mrs. Carl Oscar (Selma Augusta Emilia Johansson)'
 'Emir, Mr. Farred Chehab' 'Fortune, Mr. Charles Alexander'
 'O\'Dwyer, Miss. Ellen "Nellie"' 'Todoro

In [9]:
sex = df["Sex"].to_numpy()
print(sex)

['male' 'female' 'female' 'female' 'male' 'male' 'male' 'male' 'female'
 'female' 'female' 'female' 'male' 'male' 'female' 'female' 'male' 'male'
 'female' 'female' 'male' 'male' 'female' 'male' 'female' 'female' 'male'
 'male' 'female' 'male' 'male' 'female' 'female' 'male' 'male' 'male'
 'male' 'male' 'female' 'female' 'female' 'female' 'male' 'female'
 'female' 'male' 'male' 'female' 'male' 'female' 'male' 'male' 'female'
 'female' 'male' 'male' 'female' 'male' 'female' 'male' 'male' 'female'
 'male' 'male' 'male' 'male' 'female' 'male' 'female' 'male' 'male'
 'female' 'male' 'male' 'male' 'male' 'male' 'male' 'male' 'female' 'male'
 'male' 'female' 'male' 'female' 'female' 'male' 'male' 'female' 'male'
 'male' 'male' 'male' 'male' 'male' 'male' 'male' 'male' 'female' 'male'
 'female' 'male' 'male' 'male' 'male' 'male' 'female' 'male' 'male'
 'female' 'male' 'female' 'male' 'female' 'female' 'male' 'male' 'male'
 'male' 'female' 'male' 'male' 'male' 'female' 'male' 'male' 'male' 'ma

In [19]:
age_series = df["Age"]
print("Age Series (first 5 rows):\n", age_series.head())

Age Series (first 5 rows):
 0    22.0
1    38.0
2    26.0
3    35.0
4    35.0
Name: Age, dtype: float64


In [11]:
sib = df["SibSp"].to_numpy()
print(sib)

[1 1 0 1 0 0 0 3 0 1 1 0 0 1 0 0 4 0 1 0 0 0 0 0 3 1 0 3 0 0 0 1 0 0 1 1 0
 0 2 1 1 1 0 1 0 0 1 0 2 1 4 0 1 1 0 0 0 0 1 5 0 0 1 3 0 1 0 0 4 2 0 5 0 1
 0 0 0 0 0 0 0 0 0 0 0 3 1 0 3 0 0 0 1 1 0 0 0 0 0 1 0 0 0 0 2 0 0 0 0 1 0
 1 0 1 0 0 0 1 0 4 2 0 1 0 0 1 0 0 1 0 0 0 1 1 0 0 0 1 0 0 0 0 1 0 0 1 0 2
 0 0 0 1 0 0 0 0 0 0 0 8 0 0 0 0 4 0 0 1 0 0 0 4 1 0 0 1 3 0 0 0 8 0 4 2 0
 0 1 0 1 0 0 0 1 1 0 0 0 0 0 0 0 8 0 0 0 0 1 0 0 0 0 0 0 0 1 1 0 1 0 0 0 0
 0 0 1 0 0 0 0 3 1 0 0 4 0 0 1 0 0 0 1 1 0 0 0 2 0 0 1 1 0 1 0 1 0 0 0 0 0
 0 0 4 1 0 0 0 4 1 0 0 0 0 0 0 0 1 0 0 4 1 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0
 0 1 0 0 0 2 0 0 0 1 0 1 1 0 0 2 1 0 1 0 1 0 0 1 0 0 0 1 8 0 0 0 1 0 2 0 0
 2 1 0 1 0 0 0 1 3 0 0 0 0 0 1 1 0 0 0 1 1 0 0 0 0 0 0 1 1 0 0 1 0 1 0 0 0
 1 1 0 0 3 1 0 0 0 0 0 0 0 1 0 0 5 0 0 0 1 0 2 1 0 0 0 0 0 0 0 0 1 1 0 1 0
 1 0 3 0 0 1 0 0 0 1 0 0 0 0 0 0 1 1 0 1 0 0 0 0 1 1 0 1 1 2 2 1 0 1 0 1 0
 0 0 0 0 2 0 1 1 0 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 2 0 0 1 0 0 0 1 1 0 0 5
 0 0 0 1 3 1 0 0 1 1 0 0 

In [12]:
parch = df["Parch"].to_numpy()
print(parch)

[0 0 0 0 0 0 0 1 2 0 1 0 0 5 0 0 1 0 0 0 0 0 0 0 1 5 0 2 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 2 0 0 0 0 0 0 1 0 0 0 1 0 0 0 2 2 0 0 0 2 0 1 0 0 2 0 0 2 0 0
 0 0 0 0 2 0 0 0 0 0 0 0 3 0 2 0 0 0 0 2 0 0 0 1 1 0 0 0 1 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 1 2 0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 2 0 0 0 2 0 0 0 0 1 0 2
 2 0 0 0 0 2 0 1 0 0 0 2 1 0 0 0 1 2 1 4 0 0 0 1 1 0 0 1 1 0 0 0 2 0 2 1 2
 0 0 0 1 0 0 0 0 1 0 0 0 1 0 0 0 2 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 1 0 0 0 2 0 0 0 2 0 0 0 0 0 0 0 0 0 2 1 0 0 1 0 0 2 2 0 0 0
 1 0 2 1 0 0 0 1 0 1 0 0 0 1 1 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 2 0 1 0 0 0 0 0 2 0 0 0 0 0 2 1 0 1 0 0 0 2 1 0 0 0 1 2 0 0 0 1 1 0 0 1
 0 0 0 0 0 0 0 1 2 0 0 0 0 0 0 1 0 0 0 1 0 0 0 1 0 0 0 4 0 1 0 0 0 0 0 0 0
 0 0 0 0 1 0 0 2 0 0 0 2 0 0 0 0 2 0 0 0 2 0 0 0 2 0 0 0 0 0 0 0 0 0 0 0 0
 1 0 1 0 0 0 0 0 0 1 2 0 2 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 2 2 3 4 0 1 0 0 0
 0 2 1 0 1 0 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 2 0 0 0 0 0 0 1 2
 0 0 0 0 1 0 0 0 1 0 0 0 

In [13]:
ticket = df["Ticket"].to_numpy()
print(ticket)

['A/5 21171' 'PC 17599' 'STON/O2. 3101282' '113803' '373450' '330877'
 '17463' '349909' '347742' '237736' 'PP 9549' '113783' 'A/5. 2151'
 '347082' '350406' '248706' '382652' '244373' '345763' '2649' '239865'
 '248698' '330923' '113788' '349909' '347077' '2631' '19950' '330959'
 '349216' 'PC 17601' 'PC 17569' '335677' 'C.A. 24579' 'PC 17604' '113789'
 '2677' 'A./5. 2152' '345764' '2651' '7546' '11668' '349253'
 'SC/Paris 2123' '330958' 'S.C./A.4. 23567' '370371' '14311' '2662'
 '349237' '3101295' 'A/4. 39886' 'PC 17572' '2926' '113509' '19947'
 'C.A. 31026' '2697' 'C.A. 34651' 'CA 2144' '2669' '113572' '36973'
 '347088' 'PC 17605' '2661' 'C.A. 29395' 'S.P. 3464' '3101281' '315151'
 'C.A. 33111' 'CA 2144' 'S.O.C. 14879' '2680' '1601' '348123' '349208'
 '374746' '248738' '364516' '345767' '345779' '330932' '113059'
 'SO/C 14885' '3101278' 'W./C. 6608' 'SOTON/OQ 392086' '19950' '343275'
 '343276' '347466' 'W.E.P. 5734' 'C.A. 2315' '364500' '374910' 'PC 17754'
 'PC 17759' '231919' '244367' 

In [17]:
fare_series = df["Fare"]
print("Fare Series (first 5 rows):\n", fare_series.head())

Fare Series (first 5 rows):
 0     7.2500
1    71.2833
2     7.9250
3    53.1000
4     8.0500
Name: Fare, dtype: float64


In [20]:
print("DataFrame head():")
display(df.head())

DataFrame head():


,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [21]:
print("\nDataFrame info():")
df.info()


DataFrame info():
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [22]:
print("\nDataFrame describe():")
display(df.describe())


DataFrame describe():


,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


In [15]:
cabin = df["Cabin"].to_numpy()
print(cabin)

[nan 'C85' nan 'C123' nan nan 'E46' nan nan nan 'G6' 'C103' nan nan nan
 nan nan nan nan nan nan 'D56' nan 'A6' nan nan nan 'C23 C25 C27' nan nan
 nan 'B78' nan nan nan nan nan nan nan nan nan nan nan nan nan nan nan nan
 nan nan nan nan 'D33' nan 'B30' 'C52' nan nan nan nan nan 'B28' 'C83' nan
 nan nan 'F33' nan nan nan nan nan nan nan nan 'F G73' nan nan nan nan nan
 nan nan nan nan nan nan nan 'C23 C25 C27' nan nan nan 'E31' nan nan nan
 'A5' 'D10 D12' nan nan nan nan 'D26' nan nan nan nan nan nan nan 'C110'
 nan nan nan nan nan nan nan 'B58 B60' nan nan nan nan 'E101' 'D26' nan
 nan nan 'F E69' nan nan nan nan nan nan nan 'D47' 'C123' nan 'B86' nan
 nan nan nan nan nan nan nan 'F2' nan nan 'C2' nan nan nan nan nan nan nan
 nan nan nan nan nan nan nan 'E33' nan nan nan 'B19' nan nan nan 'A7' nan
 nan 'C49' nan nan nan nan nan 'F4' nan 'A32' nan nan nan nan nan nan nan
 'F2' 'B4' 'B80' nan nan nan nan nan nan nan nan nan 'G6' nan nan nan
 'A31' nan nan nan nan nan 'D36' nan nan 'D15'

In [16]:
embarked = df["Embarked"].to_numpy()
print(embarked)

['S' 'C' 'S' 'S' 'S' 'Q' 'S' 'S' 'S' 'C' 'S' 'S' 'S' 'S' 'S' 'S' 'Q' 'S'
 'S' 'C' 'S' 'S' 'Q' 'S' 'S' 'S' 'C' 'S' 'Q' 'S' 'C' 'C' 'Q' 'S' 'C' 'S'
 'C' 'S' 'S' 'C' 'S' 'S' 'C' 'C' 'Q' 'S' 'Q' 'Q' 'C' 'S' 'S' 'S' 'C' 'S'
 'C' 'S' 'S' 'C' 'S' 'S' 'C' nan 'S' 'S' 'C' 'C' 'S' 'S' 'S' 'S' 'S' 'S'
 'S' 'C' 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'Q' 'S' 'S' 'S' 'S' 'S' 'S' 'S'
 'S' 'S' 'S' 'S' 'S' 'S' 'C' 'C' 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'S'
 'S' 'Q' 'S' 'C' 'S' 'S' 'C' 'S' 'Q' 'S' 'C' 'S' 'S' 'S' 'C' 'S' 'S' 'C'
 'Q' 'S' 'C' 'S' 'C' 'S' 'S' 'S' 'S' 'C' 'S' 'S' 'S' 'C' 'C' 'S' 'S' 'Q'
 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'C' 'Q' 'S' 'S' 'S' 'S' 'S'
 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'Q' 'S' 'S' 'C' 'S' 'S' 'C' 'S' 'S'
 'S' 'C' 'S' 'S' 'S' 'S' 'Q' 'S' 'Q' 'S' 'S' 'S' 'S' 'S' 'C' 'C' 'Q' 'S'
 'Q' 'S' 'S' 'S' 'S' 'C' 'S' 'S' 'S' 'C' 'Q' 'C' 'S' 'S' 'S' 'S' 'Q' 'C'
 'S' 'S' 'C' 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'S' 'S'
 'S' 'S' 'S' 'S' 'S' 'S' 'C' 'Q' 'S' 'S' 'C' 'Q' 'S

In [ ]:
print("Missing values before handling:")
display(df.isnull().sum())

In [ ]:
df['Age'].fillna(df['Age'].mean(), inplace=True)
df['Embarked'].fillna(df['Embarked'].mode()[0], inplace=True)
print("Missing values after handling:")
display(df.isnull().sum())

The `Cabin` column still has a large number of missing values. Since there are too many missing values in this column, it's best to drop it. Dropping the `Cabin` column will remove it from the DataFrame.

In [ ]:
df.drop('Cabin', axis=1, inplace=True)
print("Missing values after dropping 'Cabin' column:")
display(df.isnull().sum())

In [ ]:
# Survival rate by Gender
survival_by_gender = df.groupby('Sex')['Survived'].mean().reset_index()
print("Survival Rate by Gender:")
display(survival_by_gender)

In [ ]:
# Average Age by Passenger Class
average_age_by_pclass = df.groupby('Pclass')['Age'].mean().reset_index()
print("\nAverage Age by Passenger Class:")
display(average_age_by_pclass)

In [ ]:
# Survival count by Embarked location
survival_count_by_embarked = df.groupby('Embarked')['Survived'].sum().reset_index()
print("\nSurvival Count by Embarked Location:")
display(survival_count_by_embarked)

# Task
Create a new DataFrame with `Pclass` and `ClassDescription` and merge it with the main DataFrame, then create another DataFrame with `Sex` and `GenderDescription` and join it with a subset of the main DataFrame. Finally, create a small DataFrame of new passengers and concatenate it to the main DataFrame, summarizing the outcomes of these merge, join, and concatenation operations.

## Create Additional DataFrame

### Subtask:
Create a new Pandas DataFrame containing a 'Pclass' column and a 'ClassDescription' column, which will be used for merging and joining with the main DataFrame.


**Reasoning**:
The subtask requires creating a new DataFrame `class_description_df` with 'Pclass' and 'ClassDescription' columns, populating them with specified values, and then displaying the DataFrame. This can be achieved using a single code block to define the data and construct the DataFrame.



In [23]:
data = {'Pclass': [1, 2, 3],
        'ClassDescription': ['First Class', 'Second Class', 'Third Class']}
class_description_df = pd.DataFrame(data)

print("Created Class Description DataFrame:")
display(class_description_df)

Created Class Description DataFrame:


,Pclass,ClassDescription
0,1,First Class
1,2,Second Class
2,3,Third Class


# Task
Calculate the average age of survivors versus non-survivors, find the highest and lowest fare, calculate the overall survival percentage, and summarize the identified patterns affecting survival from the Titanic dataset.

## Calculate Average Age of Survivors vs. Non-Survivors

### Subtask:
Group the DataFrame by the 'Survived' column and calculate the mean 'Age' for both groups (survivors and non-survivors). Display the result.


**Reasoning**:
The subtask requires grouping the DataFrame by the 'Survived' column and calculating the mean of the 'Age' for each group, then printing the result. I will use the `groupby()` and `mean()` methods to achieve this.



In [24]:
average_age_by_survival = df.groupby('Survived')['Age'].mean()
print("Average Age by Survival Status:")
print(average_age_by_survival)

Average Age by Survival Status:
Survived
0    30.626179
1    28.343690
Name: Age, dtype: float64


## Find Highest and Lowest Fare

### Subtask:
Find the highest and lowest fare from the 'Fare' column of the DataFrame. Display both values.

**Reasoning**:
To find the highest and lowest fare, I will use the `max()` and `min()` methods directly on the 'Fare' column of the DataFrame. This will provide the required values efficiently.



In [25]:
highest_fare = df['Fare'].max()
lowest_fare = df['Fare'].min()

print(f"Highest Fare: {highest_fare}")
print(f"Lowest Fare: {lowest_fare}")

Highest Fare: 512.3292
Lowest Fare: 0.0


## Calculate Overall Survival Percentage

### Subtask:
Calculate the overall survival percentage from the 'Survived' column of the DataFrame. Display the result as a percentage.

**Reasoning**:
To calculate the overall survival percentage, I will compute the mean of the 'Survived' column and multiply by 100, then display the result.



In [26]:
overall_survival_percentage = df['Survived'].mean() * 100
print(f"Overall Survival Percentage: {overall_survival_percentage:.2f}%")

Overall Survival Percentage: 38.38%


## Summarize Identified Patterns Affecting Survival

### Subtask:
Summarize the identified patterns affecting survival from the Titanic dataset, drawing conclusions from the previously calculated statistics (survival rate by gender, average age by survival, and overall survival percentage).

**Summary of Survival Patterns:**
Based on the analysis performed:
*   **Gender:** Females had a significantly higher survival rate compared to males (as seen in the 'survival_by_gender' calculation).
*   **Age:** The average age of survivors (around 28.34 years) was slightly lower than that of non-survivors (around 30.63 years), suggesting that younger individuals or children might have had a marginally better chance, although this difference is not large.
*   **Overall Survival:** The overall survival rate was approximately 38.38%, indicating that less than half of the passengers survived the disaster.

## Final Task

### Subtask:
Provide a summary of all calculations and identified patterns affecting survival.


## Summary:

### Q&A
1.  **What is the average age of survivors versus non-survivors?**
    The average age of survivors was approximately 28.34 years, while the average age of non-survivors was approximately 30.63 years.
2.  **What are the highest and lowest fares?**
    The highest fare paid was \$512.33, and the lowest fare paid was \$0.0.
3.  **What is the overall survival percentage?**
    The overall survival rate was 38.38%.
4.  **What are the identified patterns affecting survival?**
    Females had a significantly higher survival rate compared to males. Survivors were, on average, slightly younger than non-survivors.

### Data Analysis Key Findings
*   The average age of non-survivors was approximately 30.63 years, which is slightly higher than the average age of survivors, at approximately 28.34 years.
*   The highest fare paid by a passenger was \$512.33, while the lowest fare paid was \$0.0.
*   The overall survival rate for passengers on the Titanic was 38.38%.
*   Gender was a significant factor in survival, with females having a considerably higher survival rate than males.

### Insights or Next Steps
*   Further analysis could explore the fare distribution more deeply, especially why some fares were \$0.0, to understand its potential impact on survival or passenger class.
*   Investigate other factors like passenger class and port of embarkation, and their interaction with age and gender, to build a more comprehensive model of survival probability.
